moped-editor/src/views/projects/projectView/ProjectFiles.js

```diff
@@ -56,33 +51,225 @@ const useStyles = makeStyles((theme) => ({
   },
   codeStyle: {
     backgroundColor: "#eee",
     fontFamily: "monospace",
     display: "inline-block",
     paddingLeft: "4px",
     paddingRight: "4px",
-    fontSize: "14px"
+    fontSize: "14px",
   },
 }));
 
+const fileTypes = ["", "Funding", "Plans", "Estimates", "Other"];
+
+// remove the FilePond and s3 added path for display, ex:
+// 'private/project/65/80_04072022191747_40d4c982e064d0f9_1800halfscofieldridgepwkydesignprint.pdf'
+const cleanUpFileKey = (str) => str.replace(/^(?:[^_]*_){3}/g, "");
+
+const useColumns = ({
+  classes,
+  token,
+  rowModesModel,
+  handleEditClick,
+  handleSaveClick,
+  handleCancelClick,
+  handleDeleteOpen,
+}) =>
+  useMemo(() => {
+    return [
+      {
+        headerName: "Name",
+        field: "file_name",
+        width: 175,
+        editable: true,
+        // validate input
+        preProcessEditCellProps: (params) => {
+          const hasError = params.props.value.trim().length < 1;
+          return { ...params.props, error: hasError };
+        },
+        renderEditCell: (props) => (
+          <DataGridTextField helperText="Required" {...props} />
+        ),
+      },
+      {
+        headerName: "File",
+        field: "file_url",
+        width: 200,
+        editable: true,
+        // validate input
+        preProcessEditCellProps: (params) => {
+          const hasError = params.props.value.trim().length < 1;
+          return { ...params.props, error: hasError };
+        },
+        renderCell: ({ row }) => {
+          if (row.file_key) {
+            return (
+              <Link
+                className={classes.downloadLink}
+                onClick={() => downloadFileAttachment(row?.file_key, token)}
+              >
+                {cleanUpFileKey(row?.file_key)}
+              </Link>
+            );
+          }
+          return isValidUrl(row?.file_url) ? (
+            <ExternalLink
+              linkProps={{ className: classes.downloadLink }}
+              url={row?.file_url}
+              text={row?.file_url}
+            />
+          ) : (
+            // if the user provided file_url is not a valid url, just render the text
+            <Typography className={classes.codeStyle}>
+              {row?.file_url}
+            </Typography>
+          );
+        },
+        renderEditCell: (props) =>
+          // users cannot edit the file_key, since its provided by the FilePond upload interface
+          props.row.file_key ? (
+            <Typography>{cleanUpFileKey(props.row.file_key)}</Typography>
+          ) : (
+            <DataGridTextField
+              helperText="Required"
+              disabled={!!props.row.file_key}
+              {...props}
+            />
+          ),
+      },
+      {
+        headerName: "Type",
+        field: "file_type",
+        editable: true,
+        width: 150,
+        valueGetter: (value) => fileTypes[value],
+        renderEditCell: (props) => <ProjectFilesTypeSelect {...props} />,
+      },
```

Comment:
This input is a little squished (like half the width of the column) in edit mode. Maybe there are some tweaks needed in the Select component?
<img width="1394" alt="Screenshot 2024-10-04 at 11 47 44 AM" src="https://github.com/user-attachments/assets/9d073a1e-8643-49ab-bcb8-751411c86cb1">

I am also getting the dreaded autocomplete warning when i switch into edit mode

<img width="405" alt="Screenshot 2024-10-04 at 11 49 01 AM" src="https://github.com/user-attachments/assets/177b38a9-c497-4236-b38c-a69c4df1f284">

Reply:
I broke this when I added the valueGetter to make sorting work ☹️ . I am going to see how to fix this but also maintain ability to sort by the words and not the numbers

Reply:
I broke the select when I used valueGetter, valueFormatter changes the value only for rendering and doesn't affect the underlying value. which means you still cant sort. valueGetter affects the underlying value, which means we can sort alphabetically vs 1,2,3,4 but it was then passing the string to the select component instead of the file_type number. 

So now its valueFormatter to render the file_type strings, and passing those strings into the data grid sort comparator, while keeping the original value so the select works.

Reply:
ditto—great to have these details ironed out for the next MaterialTable that gets replaced ✅